sensor-gateway/common/error/SensorAccessLinkError.cpp

```diff
@@ -0,0 +1,161 @@
+    auto messagesAreEqual = (sameOrigin &&
+                             sameCategory &&
+                             sameSeverity &&
+                             sameCode &&
+                             sameMessage
+    );
+    return messagesAreEqual;
+}
+
+bool SensorAccessLinkError::operator!=(SensorAccessLinkError const& other) const {
+    return !operator==(other);
+}
+
+SensorAccessLinkError const SensorAccessLinkError::returnDefaultData() noexcept {
+    return Defaults::DEFAULT_SENSOR_ACCESS_LINK_ERROR;
+}
+
+bool SensorAccessLinkError::isFatalForGateway() const noexcept {
+    bool isFatal = false;
+    if (severity == EMERGENCY) {
+        isFatal = true;
+    }
+    return isFatal;
+}
+
+bool SensorAccessLinkError::isFatalForSensorAccess() const noexcept {
+    bool isFatal = false;
+    if (severity == EMERGENCY ||
+        severity == ALERT) {
+        isFatal = true;
+    }
+    return isFatal;
+}
+
+bool SensorAccessLinkError::isOpenConnectionRequired() const noexcept {
+    bool isOpenConnectionRequired = true;
+    if (category == COMMUNICATION_ERROR ||
+        severity == EMERGENCY ||
+        severity == ALERT) {
+        isOpenConnectionRequired = false;
+    }
+    return isOpenConnectionRequired;
+}
+
+bool SensorAccessLinkError::isCloseConnectionRequired() const noexcept {
+    bool isCloseConnectionRequired = true;
+    if (category == COMMUNICATION_ERROR) {
+        isCloseConnectionRequired = false;
+    }
+    return isCloseConnectionRequired;
+}
+
+std::string SensorAccessLinkError::fetchDetailedMessage() const noexcept {
+    std::string detailedMessage = buildDetailedMessage(timestamp, origin, category, severity, code, message);
+    return detailedMessage;
+}
+
+std::string SensorAccessLinkError::buildDetailedMessage(HighResolutionTimePoint const& timestamp,
+                                                        std::string const& origin,
+                                                        Category const& category,
+                                                        Severity const& severity,
+                                                        ErrorCode const& code,
+                                                        std::string const& message) noexcept {
+    auto formattedTimestamp = formatTimestamp(timestamp);
+    std::ostringstream detailedMessageStream(formattedTimestamp, std::ios_base::ate);
+    detailedMessageStream
+            << " -- <" << severity
+            << "> " << category
+            << " - " << origin
+            << ": " << code
+            << " ->  " << message
+            << "\n";
+    std::string detailedMessage = detailedMessageStream.str();
+    return detailedMessage;
+}
+
+std::string SensorAccessLinkError::formatTimestamp(HighResolutionTimePoint const& timestamp) noexcept {
+    std::size_t const ARBITRARY_BIG_ENOUGH_SIZE = 128;
+    auto time_tTimestamp = std::chrono::_V2::system_clock::to_time_t(timestamp);
+    auto gmTime = gmtime(&time_tTimestamp);
```

Comment:
These are synonyms, I don't get the purpose of your comment?

Reply:
GMT is a deprecated timezone that no longer exists and is now called UTC. Plus, I had to wonder when I saw gmTime what it was (compared to GMTTime, which would have been clearer).

It's a minor case of preference, but I mentioned it mostly for the deprecation. Since you already name the variable, might as well name it in an up-to-date and clear way.